desc/geometry/curve.py

```diff
@@ -1218,3 +1289,440 @@ def compute_length(self, center=None, normal=None, r_n=None, grid=None):
         T = jnp.linalg.norm(T, axis=1)
         theta = transform.grid.nodes[:, 2]
         return jnp.trapz(T, theta)
+
+
+class XYZCurve(Curve):
```

Comment:
I'd prefer if this was `SplineXYZCurve` for consistency with the other names

Reply:
I will change this before the PR is finalized